lib/note-content-editor.tsx

```diff
@@ -150,8 +152,96 @@ class NoteContentEditor extends Component<Props> {
         }
       }, 400);
     }
+
+    if (this.editor && prevProps.searchQuery !== this.props.searchQuery) {
+      this.setDecorators();
+    }
   }
 
+  setDecorators = () => {
+    const searchMatches = this.searchMatches();
+    const titleDecoration = this.getTitleDecoration();
+    if (!searchMatches.length && !titleDecoration) {
+      return;
+    }
+    if (titleDecoration) {
+      searchMatches.push(titleDecoration);
+    }
+    decorations = this.editor.deltaDecorations(decorations, [...searchMatches]);
+  };
+
+  getTitleDecoration = () => {
+    const model = this.editor.getModel();
+    if (!model) {
+      return;
+    }
+    const titleDecoration = (line: number) => ({
+      range: new monaco.Range(line, 1, line, 1),
+      options: {
+        isWholeLine: true,
+        inlineClassName: 'note-title',
+        stickiness: Editor.TrackedRangeStickiness.NeverGrowsWhenTypingAtEdges,
+      },
+    });
```

Comment:
as this is a static function we can move it into the module global scope above `class NoteContentEditor`

won't make a meaningful impact but it will do less than defining it here does in the VM and it will keep this function smaller. maybe we should consider a rename: `titleDecorationForLine(line: number)`